doc/hotfix-process.md

```diff
@@ -0,0 +1,54 @@
+Hotfix Release Process
+======================
+
+## Create a hotfix branch
+
+Crete a hotfix branch from the previous release tag, and push it to the main
+repository:
+
+    $ git branch hotfix-<RELEASE> <RELEASE_PREV>
+    $ git push 'git@github.com:zcash/zcash' hotfix-<RELEASE>
+
+## Merge hotfix PRs
+
+Developer should create hotfix PRs based on the hotfix branch. Each PR should be
+reviewed as normal, and then the following process should be used to merge:
+
+- A CI merge build is manually run, by force-triggering the pr-merge builder
+  with the following fields set:
+
+  - Repository: https://github.com/<DevUser>/zcash
+    - <DevUser> must be in the set of "safe" users as-specified in the CI
+      config.
+  - Branch: name of the hotfix PR branch (not the hotfix release branch).
+
+- A link to the build and its result is manually added to the PR as a comment.
+
+- If the build was successful, the PR is merged via the GitHub button.
+
+## Release process
+
+The majority of this process is identical to the standard release process.
+However, there are a few notable differences:
+
+- When running the release script, use the `--hotfix` flag:
+
+    $ ./zcutil/make-release.py --hotfix <RELEASE> <RELEASE_PREV> <APPROX_RELEASE_HEIGHT>
+
+- To review the automated changes in git:
+
+    $ git log hotfix-<RELEASE>..HEAD
+
+- After the standard review process, use the hotfix merge process outlined above
+  instead of the regular merge process.
+
+- When making the tag, check out the hotfix branch instead of master.
+
+## Post-release
+
+Once the hotfix release has been created, a new PR should be opened for merging
+the hotfix release branch into master. This may require fixing merge conflicts
+(e.g. changing the version number in the hotfix branch to match master, if
+master is ahead). Such conflicts **MUST** be addressed with additional commits
+to the hotfix branch; specifically, the branch **MUST NOT** be rebased on
+master.
```

Comment:
The actual release revision of a hotfix will include a signed release tag on the hotfix branch, correct? So if _further_ commits are necessary to fix merge conflicts, it will be explicitly visible which changes were included in the release versus those that were required for merging back into master, correct?

Also, does this guarantee that all releases, even hotfix releases will (eventually) be ancestors of the latest `master`? (I recall some weird behavior with tags where only tags on ancestors of the current branch are fetched by default, or some other weird edge, and I want to ensure whoever pulls `master` gets all our release tags.)

Reply:
> The actual release revision of a hotfix will include a signed release tag on the hotfix branch, correct? So if further commits are necessary to fix merge conflicts, it will be explicitly visible which changes were included in the release versus those that were required for merging back into master, correct?

Correct. This is why I have the last line in here - rebasing the hotfix branch would destroy this link (at least cryptographically, and possibly entirely if commits get squashed or edited).

> Also, does this guarantee that all releases, even hotfix releases will (eventually) be ancestors of the latest master? (I recall some weird behavior with tags where only tags on ancestors of the current branch are fetched by default, or some other weird edge, and I want to ensure whoever pulls master gets all our release tags.)

As long as the hotfix branch is subsequently merged into master, yes (to my knowledge). If we wanted to be more explicit/clear or "sure" about this, we'd need to switch to a development model where every release corresponds to a commit to master, which could be something along [these lines](http://nvie.com/posts/a-successful-git-branching-model/).